### ISSUE TITLE
LibriTTS-460 training text And Training result

### ISSUE THREAD

**Issue by WendongGan** [yl4579/StyleTTS2#95]
@yl4579 Hi, yl4579,I have a doubt. For LibriTTS-460 training text, you only gave examples from styletts1, and did not provide train_libritts_list.txt and val_libritts_list.txt of styletts2. But there is no speaker id in the example of styletts1. I want to confirm one question, for example: LibriTTS/train-clean-460/2113/132485/2113_132485_000061_000006.wav||aɪ wʌz kɹˈuːəl!

Did you train 2113_132485 as a speaker or 2113 as a speaker? At present, I use 2113 as a speaker, that is, I use the audio of different chapters of this speaker as a speaker. So far, I've been training from scratch, following your code, stage1 and stage2. config parameters, diff_epoch: 10, joint_epoch: 15, my stage2 has been trained to 25epoch, still worse than your pre-trained model (like 20epcoh, diff_epoch: 0, joint_epoch: 0), I'm wondering where I might be at odds with you.  

 Look forward your help.  If possible, can you upload your train_libritts_list.txt and val_libritts_list.txt ?

**Comment by WendongGan** [comment]
This is my eval loss.  
<img width="668" alt="image" src="https://github.com/yl4579/StyleTTS2/assets/40649244/980638d5-389b-470b-8127-cf524bb3fa91">


As for the train loss, it is a bit of a pity that my training was stopped by me at 25epoch yesterday. I continued the training from 25epcoh, but I looked in the code and the iters were set to 0, so the tensorboard I found was overwritten, only the most recent ones.

<img width="478" alt="image" src="https://github.com/yl4579/StyleTTS2/assets/40649244/3af6c6c6-063b-4433-8626-1d57ce365bd9">

My train loss is :

<img width="652" alt="image" src="https://github.com/yl4579/StyleTTS2/assets/40649244/e95dd100-9c77-4280-96f9-f3866e3c2cdd">

**Comment by yl4579** [comment]
It doesn't look abnormal at all. What kind of audio did you get out of it?

**Comment by WendongGan** [comment]
In addition, I continued the training on your pre-trained model (epochs_2nd_00020.pth) and I still set diff_epoch: 0, joint_epoch: 0. Here is my training curve.

<img width="649" alt="image" src="https://github.com/yl4579/StyleTTS2/assets/40649244/6c2e77ee-f820-429e-ab61-35bdcfdc57cf">

<img width="697" alt="image" src="https://github.com/yl4579/StyleTTS2/assets/40649244/473c3e32-cc0d-42d0-9013-bea5a0b341e4">

I found the eval mel_loss is only 0.37, but the one trained by myself is  still 0.385.

**Comment by yl4579** [comment]
By the way, I found a bug in the code. The `dur_loss` and `mel_loss` is exactly the same for the evaluation loss. 

I think looking at mel loss doesn't tell you a lot, but all your loss seems higher than what I got (including F0 loss). This says something might be off during training. What kind of speech sample did you get in the end?

**Comment by WendongGan** [comment]
These is the sample generated by the model trained by mayself and your pretrained model.
https://drive.google.com/file/d/1vq4AWtX02tDuNsIpvvOCJqj2-5uJm-Ds/view?usp=drive_link

**Comment by yl4579** [comment]
I have no access to the files.

**Comment by WendongGan** [comment]
sorry，I have modified it.

**Comment by WendongGan** [comment]
> I have already uploaded the training data, which is in https://github.com/yl4579/StyleTTS2/blob/main/Data/OOD_texts.txt, including the speaker ids. I used the training texts from LibriTTS as the OOD texts as described in the paper.
> 
> Could you show me the training curve? The diff_epoch and joint_epoch setting to 0 was because I had to resume training so I just set them to a low number. They aren’t the actual epochs those training started.

I checked the text, My understanding and what i do is correct, that is, 2113 is regarded as a speaker, that is, the audio of different chapters from the same speaker is regarded as a speaker.

**Comment by yl4579** [comment]
@WendongGan I listened to the samples you sent me. I think the quality isn't that different, and I wouldn't say one model is definitely better than the other in terms of naturalness. Are the reference speakers from LibriTTS or not? Are they seen speakers or not?

What's interesting is that the speaker `0011` speaks in British accent, and the pre-trained model could capture this yet your model couldn't. Did you use the latest code that has the `st` fixed?

**Comment by WendongGan** [comment]
@yl4579 The test sample is from [ESD](https://github.com/HLTSingapore/Emotional-Speech-Data) . it is not from LibriTTS.  
Yes, I also think just the prosody of my sample is less natural than yours. 

the Code of train_second.py I used is [this one](https://github.com/yl4579/StyleTTS2/tree/2f2dece35b9cad63215ae38bf991b856d1d1c063) .

What I find different is [here](https://github.com/yl4579/StyleTTS2/commit/e2fabb339f14db62d141f73d89bed134e5ec4ebc)

The pre-training model you provided is really good, and the effect of my training is gradually approaching. What I want to verify is whether the current code can efficiently and quickly achieve the effect of your pre-trained model. When I do, my improvement attempts and other work may be more meaningful.I am trying the new code.   Do you have any other advice or share?

**Comment by yl4579** [comment]
@WendongGan This is one of the mistakes but I don't think it'll make such a big difference. The model was produced using the code slightly different from the current version (mostly https://github.com/yl4579/StyleTTS2/commit/32f760b4a558b683aa225a0129f0db7cda8d100e this version with change here: https://github.com/yl4579/StyleTTS2/commit/041ea6d3f1cc08324a16c955608a22ae8e0dcb39). I don't think I have time to verify whether the current code can reproduce the pre-trained model quality, but I don't think the quality would change that much. Also the epoch number is not exactly 20 or anything because I reset the number and stopped it manually (it was only named as epoch 20). I think the best strategy is to listen to the samples periodically and stop at one that sounds the best because the SLM adversarial training could get unstable sometimes, but your probably seems deeper than that as the accent was not captured either, so it doesn't sound like a problem of SLM adversarial training. Maybe some earlier epochs should sound better?
